app.py

```diff
@@ -0,0 +1,423 @@
+                if not task_template:
+                    raise TankError("The task template '%s' specified in the task_template setting "
+                                    "does not exist!" % task_template_name)
+                
+            sg_data = self.shotgun.create(entity_type, {"code": entity_name, 
+                                                        "description": "Created by the Shotgun Flame integration.",
+                                                        "task_template": task_template,
+                                                        "project": self.context.project})
+            
+            self.log_debug("Created %s" % sg_data)
+            
+            # now extract and upload a thumbnail
+            # for the extraction, we use the readframe utility which is part of the wiretap library
+            # Syntax:
+            #
+            # Usage: ./read_frame
+            #   -n <clip node id> (if empty, generate 4x4 black media)
+            #   [ -h <Wiretap server ID> (default = localhost) ]
+            #   [ -W <display width> (default=same as source) ]
+            #   [ -H <display height> (default=same as source) ]
+            #   [ -b <output bits per pixel (24|32)> (default = 24) ]
+            #   [ -i <zero-based start frame idx> (default = 0) ]
+            #   [ -N <number of frames to output> (default = 1, -1 for all)
+            #   [ -r (output raw RGB, default=jpg) ]
+            #   [ -O (flip raw output orientation, default=bottom to top) ]
+            #   [ -L (use lowest resolution available, default=highest) ]
+            #   [ -c <compression factor [0,100]> (default = 100)
+            #   [ -p <processing options> (default = none)
+            #
+            input_cmd = "%s -n \"%s@CLIP\" -h %s -W %s -H %s -L" % (self.engine.get_read_frame_path(),
+                                                                    full_path,
+                                                                    "%s:Gateway" % self.engine.get_server_hostname(), 
+                                                                    info["width"],
+                                                                    info["height"])
+            
+            thumbnail_jpg = os.path.join(self.engine.get_backburner_tmp(), "tk_thumb_%s.jpg" % uuid.uuid4().hex)
+            full_cmd = "%s > %s" % (input_cmd, thumbnail_jpg)
+            self.log_debug("Executing %s" % full_cmd)
+            if os.system(full_cmd) != 0:
+                self.log_warning("Could not extract thumbnail! See error log for details.")
+            else:
+                self.log_debug("Wrote thumbnail %s" % thumbnail_jpg)
+                self.sgtk.shotgun.upload_thumbnail(sg_data["type"], sg_data["id"], thumbnail_jpg)
+                self.log_debug("Uploaded thumbnail to shotgun.")
+                try:
+                    os.remove(thumbnail_jpg)
```

Comment:
Should this be in a finally clause, so thumbnail file is always deleted ? What is the reason to have the remove call in a try clause ? Is it in case it was not created, or it can it be deleted by something else ?